test/functional/test_framework/test_framework.py

```diff
@@ -206,6 +208,8 @@ def parse_args(self):
         config = configparser.ConfigParser()
         config.read_file(open(self.options.configfile))
         self.config = config
+        if self.options.v1transport:
+            self.options.v2transport=False
```

Comment:
v1 overriding v2 is somewhat non intuitive (v2 overriding v1 is as well). What we really want here is a local option overriding a global one, regardless of which one is v1 and v2. Can we have this somehow? Maybe have `test_runner.py` when ran like e.g. `test_runner.py --v1` and when picking entries from `BASE_SCRIPTS[]` to check that an entry like `p2p_timeouts.py --v2` already contains a conflicting option and omit that entry altogether?

Also, it would be best if an individual test is ran like `test/functional/p2p_fingerprint.py --v1 --v2` to emit an error and refuse to start (e.g. "conflicting options given") instead of running in `v1` mode.